libnmstate/state.py

```diff
@@ -30,6 +32,31 @@
 from libnmstate.prettystate import format_desired_current_state_diff
 from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceType
+from libnmstate.schema import Route
+from libnmstate.nm import route as nm_route
+
+
+class _Route(object):
+    def __init__(self, route):
+        self.state = {
```

Comment:
I assumed this is the sub-state for just a route entry, so it will need to be created anyhow.

Reply:
Thanks for the clarification.
Processing data using attributes seems to be more appropriate to me. The way the data is represented is a different manner.
Note that using dicts will also cause the data to be shared by two sources (one dict, multiple pointers to it). I do not know the implications of that in this case. I just found it surprising and that is why I asked.

Reply:
@tyll, you are right about the reference, no idea what I was thinking at the time.
I guess it ends up to the question of how data should be preserved in the object.
I specifically do not think this is a nice state representation, instead of referencing the specific attribute, there is a need to look it up in the dict. The hashing as an example, needs to dig in the dict instead of listing the attributes...
See https://github.com/nmstate/nmstate/pull/345/files#diff-cf08a4a46976f25e68653866cd783ebbR26